tests/testAutopas/tests/selectors/AutoTunerTest.cpp

```diff
@@ -7,75 +7,167 @@
 
-  EXPECT_EQ(i, 4 * numSamples + 1) << "Unexpected number of tuning iterations!";
+  configTimes[bestConfig] = {2, 3, 3, 100};
+  configTimes[autopas::Configuration(autopas::ContainerOption::directSum, autopas::TraversalOption::directSumTraversal,
+                                     autopas::DataLayoutOption::aos, autopas::Newton3Option::disabled)] = {4, 1, 5, 2};
 
-  auto container = autoTuner.getContainer();
-  EXPECT_EQ(fastestContainer->getContainerType(), container->getContainerType())
-      << "tune() returned the wrong container after tuning phase";
+  mapConfigTime ignoredConfigTimes;
+  ignoredConfigTimes[autopas::Configuration(autopas::ContainerOption::directSum,
+                                            autopas::TraversalOption::directSumTraversal,
+                                            autopas::DataLayoutOption::soa, autopas::Newton3Option::disabled)] = {1, 5};
+
+  testFastest(autopas::SelectorStrategy::fastestMedian, configTimes, bestConfig, ignoredConfigTimes);
 }
 
-TEST_F(AutoTunerTest, testTuneAoS) { testTune(autopas::DataLayoutOption::aos); }
+void AutoTunerTest::testFastest(autopas::SelectorStrategy strategy, mapConfigTime configAndTimes,
+                                autopas::Configuration expectedBest, mapConfigTime ignoredConfigAndTimes) {
+  unsigned int maxSamples = 4;
+
+  std::set<autopas::Configuration> uniqueConfigs;
+
+  std::set<autopas::ContainerOption> containerOptions;
+  std::set<autopas::TraversalOption> traversalOptions;
+  std::set<autopas::DataLayoutOption> dataLayoutOptions;
+  std::set<autopas::Newton3Option> newton3Options;
+
+  for (auto &&m : configAndTimes) {
+    uniqueConfigs.insert(m.first);
+  }
+  for (auto &&m : ignoredConfigAndTimes) {
+    uniqueConfigs.insert(m.first);
+  }
+  for (auto &&c : uniqueConfigs) {
+    containerOptions.insert(c._container);
+    traversalOptions.insert(c._traversal);
+    dataLayoutOptions.insert(c._dataLayout);
+    newton3Options.insert(c._newton3);
+  }
+
+  constexpr size_t domainSize = 1000;
+
+  autopas::AutoTuner<Particle, FPCell> tuner(
+      {0, 0, 0}, {domainSize, domainSize, domainSize}, 1, 0, 100,
+      std::vector<autopas::ContainerOption>(containerOptions.begin(), containerOptions.end()),
+      std::vector<autopas::TraversalOption>(traversalOptions.begin(), traversalOptions.end()),
+      std::vector<autopas::DataLayoutOption>(dataLayoutOptions.begin(), dataLayoutOptions.end()),
+      std::vector<autopas::Newton3Option>(newton3Options.begin(), newton3Options.end()), strategy, 1000, maxSamples);
+
+  // check that all expected configurations are created
+  ASSERT_EQ(uniqueConfigs.size(), tuner.getAllowedConfigurations().size());
+
+  MFunctor functor;
+
+  for (auto &&m : configAndTimes) {
+    for (auto &&t : m.second) {
+      EXPECT_CALL(functor, isRelevantForTuning()).WillOnce(::testing::Return(true));
+      tuner.addTimeMeasurement(functor, t);
```

Comment:
how can this even work?
you are not incrementing the _currentConfig of the tuner...

Reply:
-> make new class for strategies + measurements